src/misc.jl

```diff
@@ -201,3 +201,42 @@ end
     root --> :top
     GraphPlot((source, destiny))
 end
+
+
+@userplot AndrewsPlot
+
+"""
+https://en.wikipedia.org/wiki/Andrews_plot
+
+Example
+```julia
+using RDatasets, PlotRecipes
+iris   = dataset("datasets", "iris")
+y      = Array(iris[:,1:4])
+groups = Array(iris[:,5])
+andrewsplot(y, groups)
+```
+"""
+@recipe function f(h::AndrewsPlot)
+    y = h.args[1]
+    seriestype --> :path
+    title --> "Andrew's plot"
+    label := ""
+    rows,cols = size(y)
+    serieslength = 100
+    ys    = zeros(serieslength,rows)
+    t     = linspace(-π,π,serieslength)
+    for j in 1:rows, ti = eachindex(t)
+        ys[ti,j] = y[j,1]/sqrt(2) + sum(y[j,i].*sin.((i÷2).*t[ti]) for i = 2:cols)
+    end
+    if length(h.args) >= 2
```

Comment:
Plots deals with groups by making them `series`. It would be confusing to special-case groups as a second positional argument here. What is really needed is to edit the way Plots works with groups, as it causes problems for the more advanced uses of groups here in StatPlots (as also experienced by `groupapply`).
Current suggestion: specify a keyword argument `group` and use that to split the data. Rather than just changing the color, cycle over unique values of `group` and specify new series with the `@series` macro. The colors will be handled automatically behind the scenes by Plots.